hawc/apps/vocab/api.py

```diff
@@ -83,3 +85,44 @@ def related_entity(self, request: Request, pk: int = None) -> Response:
         entity.terms.add(term, through_defaults={"notes": request.data.get("notes", "")})
         serializer = self.get_serializer(entity.terms.all(), many=True)
         return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class TermViewset(mixins.CreateModelMixin, viewsets.GenericViewSet):
+    serializer_class = serializers.TermSerializer
+    queryset = models.Term.objects.all()
+
+    @transaction.atomic
+    @action(
+        detail=False, methods=("post",), permission_classes=(IsAdminUser,), url_path="bulk-create"
+    )
+    def bulk_create(self, request: Request) -> Response:
+        # ensure data is list of dicts
+        try:
```

Comment:
can we move this try/except to a helper method for this viewset for both bulk_createa and bulk_update?  and i think it should say "data must be an array of objects" since that's more JSON-specific than dicts in python.

On a related note; is this needed? I think lots of this logic is in the init of the BulkSerializer

Reply:
`ListSerializer` does check if its a list and throws a `ValidationError` if it isn't, but if I were to pass it a list of ints or some other invalid list I think it would result in a 500 error. I'll reconfirm in a moment after some quick testing

Reply:
ok, if it's needed keep it for sure, just wanted to make sure. please use a single shared method though for both the update and the create

Reply:
Oops, it looks like it was an error of my own creation; when I copied the inherited method verbatim, I didn't take into account that `ValidationError` in `common\serializers.py` was the Django exception, so the inherited method was throwing that instead of the DRF error. Its fixed now and is thrown correctly from the serializer, so I should be able to remove this from `api.py`!